lib/rust/parser/src/syntax/operator/types.rs

```diff
@@ -129,36 +147,155 @@ impl Warnings {
 // ======================================
 
 pub trait OperandConsumer<'s> {
-    fn push_operand(&mut self, operand: Operand<Tree<'s>>);
+    fn push_operand(&mut self, operand: MaybeSection<Tree<'s>>);
 }
 
 pub trait OperatorConsumer<'s> {
     fn push_operator(&mut self, operator: Operator<'s>);
 }
 
+pub trait NamedOperandConsumer<'s> {
+    fn push_maybe_named_operand(&mut self, operand: OperandMaybeNamed<'s>);
+}
 
-// ===========================
-// === Operator or Operand ===
-// ===========================
 
-#[derive(Debug)]
-pub enum OperatorOrOperand<'s> {
-    Operand(Operand<Tree<'s>>),
-    Operator(Operator<'s>),
+// === Debugging ===
+
+#[derive(Debug, Default)]
+pub struct Inspect<Inner>(Inner);
+
```

Comment:
Is this `Inspect` a debugging tool for operators only (what it's placement would imply) or for general use (as the trait impls seem to me quite general)? If the latter, I would place it somewhere else.